### PR TITLE
Reduce duplication between ssl_msg.c and ssl_tls13_messaging.c, #2

### PR DESCRIPTION
**Pull request by hanno-becker**
This is a continuation of #66 - see the description there.

### CODE REVIEW

File: library/ssl_tls13_client.c [hunk 1]
```diff
@@ -2390,7 +2392,7 @@ static int ssl_certificate_request_coordinate( mbedtls_ssl_context* ssl )
 #else
     if( ssl->keep_current_message == 0 )
     {
-        if( ( ret = mbedtls_ssl_read_record( ssl ) ) != 0 )
+        if( ( ret = mbedtls_ssl_read_record( ssl, 1 ) ) != 0 )
```
Comment (hannestschofenig) [inline library/ssl_tls13_client.c]: Why didn't you want to put the SSL_FORCE_FLUSH here (and below as well)?

Comment (hanno-becker) [reply library/ssl_tls13_client.c]: Hm, good question, maybe because it wasn't defined in this file? I'll check.

File: library/ssl_msg.c [hunk 2]
```diff
@@ -3525,6 +3525,14 @@ static int ssl_check_record_type( uint8_t record_type )
...
 }
 
+static int ssl_parse_record_header( mbedtls_ssl_context const *ssl,
+                                    unsigned char *buf,
+                                    size_t len,
+                                    mbedtls_record *rec )
+{
+    /* Check TLS version */
+}
+
```
Comment (hannestschofenig) [inline library/ssl_msg.c]: Why is this code snippet here (with an empty function implementation) when the same function is implemented a few lines below?

Comment (hanno-becker) [reply library/ssl_msg.c]: That's a mistake.

Comment (hanno-becker) [inline library/ssl_msg.c]:
```suggestion
```

File: library/ssl_tls13_client.c (hunk 1 above)
Comment (hanno-becker) [inline library/ssl_tls13_client.c]:
```suggestion
        if( ( ret = mbedtls_ssl_read_record( ssl, SSL_FORCE_FLUSH ) ) != 0 )
```

**Pull request merged**